url/src/lib.rs

```diff
@@ -2720,17 +2720,25 @@ impl Url {
                 _ => return Err(()),
             };
 
-            let estimated_capacity = self.as_str().len()
-                - if cfg!(target_os = "redox") {
-                    // remove only // because it still has file:
-                    2
-                } else if cfg!(windows) {
-                    // remove file: - has posssible \\ for hostname
-                    5
+            let str_len = self.as_str().len();
+            let estimated_capacity = if cfg!(target_os = "redox") {
+                let scheme_len = self.scheme().len();
+                let file_scheme_len = "file".len();
+                // remove only // because it still has file:
+                if scheme_len < file_scheme_len {
+                    let scheme_diff = file_scheme_len - scheme_len;
```

Comment:
This still wraps around when `scheme` is longer than 4.

Reply:
There's a check above for this:

```
if scheme_len < file_scheme_len { // if 5 < 4 {
```